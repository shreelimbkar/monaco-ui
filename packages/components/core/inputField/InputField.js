"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var InputFieldStyles_1 = require("./InputFieldStyles");
var InputField = function (props) {
    var children = props.children;
    var ref = react_1.default.useRef(null);
    return (react_1.default.createElement(InputFieldStyles_1.InputFieldStyles, __assign({ as: "input", ref: ref }, props), children));
};
exports.default = InputField;
