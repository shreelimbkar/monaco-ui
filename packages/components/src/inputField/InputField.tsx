import React from 'react';

import InputProps from '../types/InputProps';
import { InputFieldStyles } from './InputFieldStyles';

const InputField: React.FC<InputProps> = (props) => {
  const { children } = props;
  const ref = React.useRef(null);
  return (
    <InputFieldStyles as="input" ref={ref} {...props}>
      {children}
    </InputFieldStyles>
  );
};

export default InputField;
