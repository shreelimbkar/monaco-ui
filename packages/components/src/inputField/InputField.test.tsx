import React from 'react';
import { render, cleanup } from '@testing-library/react';
import InputField from '.';

afterEach(cleanup);

describe('InputField Component', () => {
  test('Renders Correctly', () => {
    const { asFragment } = render(<InputField />);
    expect(asFragment()).toMatchSnapshot();
  });
});
