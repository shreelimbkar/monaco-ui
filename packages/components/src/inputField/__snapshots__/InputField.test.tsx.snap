// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Component Renders Correctly 1`] = `
<DocumentFragment>
  <input
    class="sc-bdvvtL gwXGkL"
  />
</DocumentFragment>
`;
